exchanges/binanceus/binanceus.go

```diff
@@ -0,0 +1,1966 @@
+		return nil, errIncompleteArguments
+	}
+	params.Set("symbol", strings.ToUpper(arg.Symbol))
+	if arg.OrderID > 0 {
+		params.Set("orderId", strconv.Itoa(int(arg.OrderID)))
+	}
+	timestamp := time.Now().UnixMilli()
+	params.Set("timestamp", strconv.Itoa(int(timestamp)))
+	if arg.OrigClientOrderID != "" {
+		params.Set("origClientOrderId", arg.OrigClientOrderID)
+	}
+	if arg.recvWindow > 200 && arg.recvWindow <= 6000 {
+		params.Set("recvWindow", strconv.Itoa(int(arg.recvWindow)))
+	}
+	return &resp, bi.SendAuthHTTPRequest(ctx,
+		exchange.RestSpotSupplementary,
+		http.MethodGet, orderRequest,
+		params, spotDefaultRate,
+		&resp)
+}
+
+// GetAllOpenOrders to get all open trade orders on a token symbol. Do not access this without a token symbol as this would return all pair data.
+func (bi *Binanceus) GetAllOpenOrders(ctx context.Context, symbol string) ([]*Order, error) {
+	var response []*Order
+	params := url.Values{}
+
+	timestamp := time.Now().UnixMilli()
+	if symbol != "" {
+		params.Set("symbol", symbol)
+	}
+	params.Set("timestamp", strconv.Itoa(int(timestamp)))
+	params.Set("recvWindow", recvWindowSize5000String)
+	var rateLimit request.EndpointLimit
+	if symbol != "" {
+		rateLimit = spotOpenOrdersSpecificRate
+	} else {
+		rateLimit = spotOpenOrdersAllRate
+	}
+	return response, bi.SendAuthHTTPRequest(ctx,
+		exchange.RestSpotSupplementary, http.MethodGet,
+		openOrders, params,
+		rateLimit, &response)
+}
+
+// CancelExistingOrder to cancel an active trade order.
+func (bi *Binanceus) CancelExistingOrder(ctx context.Context, arg *CancelOrderRequestParams) (*Order, error) {
+	params := url.Values{}
+	var response Order
+	params.Set("timestamp", strconv.FormatInt(time.Now().UnixMilli(), 10))
+	symbolValue, err := bi.FormatSymbol(arg.Symbol, asset.Spot)
+	if err != nil || symbolValue == "" {
+		return nil, errMissingCurrencySymbol
+	}
+	params.Set("symbol", symbolValue)
+	if arg.OrderID == "" && arg.ClientSuppliedOrderID == "" {
+		return nil, errEitherOrderIDOrClientOrderIDIsRequired
+	}
+	if arg.ClientSuppliedOrderID != "" {
+		params.Set("origClientOrderId", arg.ClientSuppliedOrderID)
+	} else {
+		params.Set("orderId", arg.OrderID)
+	}
+	params.Set("recvWindow", recvWindowSize5000String)
+	return &response, bi.SendAuthHTTPRequest(ctx,
+		exchange.RestSpotSupplementary,
+		http.MethodDelete, orderRequest,
+		params, spotDefaultRate, &response)
+}
+
+// CancelOpenOrdersForSymbol request to cancel an open orders.
+func (bi *Binanceus) CancelOpenOrdersForSymbol(ctx context.Context, symbol string) ([]Order, error) {
+	params := url.Values{}
+	if symbol == "" || len(symbol) < 4 {
+		return nil, errMissingCurrencySymbol
+	}
+	params.Set("symbol", symbol)
+	params.Set("timestamp", strconv.Itoa(int(time.Now().UnixMilli())))
+	params.Set("recvWindow", "5000")
+	var response []Order
+	return response, bi.SendAuthHTTPRequest(ctx, exchange.RestSpotSupplementary,
+		http.MethodDelete, openOrders,
+		params, spotDefaultRate, response)
```

Comment:
```suggestion
		params, spotOrderRate, response)
```

Reply:
have you noticed that the rate limit is changing with the availability of the symbol